publish-mqtt/src/main.rs

```diff
@@ -272,17 +349,44 @@ async fn connect_first_sensor_in_queue(
 }
 
 async fn connect_start_sensor<'a>(
-    bt_session: &'a BluetoothSession,
+    dbus_conn: Arc<SyncConnection>,
     homie: &mut HomieDevice,
     sensor: &mut Sensor,
-) -> Result<(), Box<dyn Error>> {
-    let device = sensor.device(bt_session);
-    device.connect(CONNECT_TIMEOUT_MS)?;
-    start_notify_sensor(bt_session, &device)?;
-
-    homie.add_node(sensor.as_node()).await?;
-    sensor.last_update_timestamp = Instant::now();
-    Ok(())
+) -> Result<(), anyhow::Error> {
+    let device = sensor.device(dbus_conn.clone());
+    println!("Connecting from status: {:?}", sensor.connection_status);
+    device
+        .connect()
+        .await
+        .with_context(|| std::line!().to_string())?;
+    match start_notify_sensor(dbus_conn.clone(), &sensor.object_path).await {
+        Ok(()) => {
+            homie
+                .add_node(sensor.as_node())
+                .await
+                .with_context(|| std::line!().to_string())?;
+            sensor.connection_status = ConnectionStatus::Connected;
+            sensor.last_update_timestamp = Instant::now();
+            Ok(())
+        }
+        Err(e) => {
+            // If starting notifications failed, disconnect so that we start again from a clean
+            // state next time.
```

Comment:
- [ ] make a task to drive this off the services_resolved property changed signal